src/Umbraco.Web/Editors/LanguageController.cs

```diff
@@ -99,24 +99,28 @@ public Language SaveLanguage(Language language)
                 throw new HttpResponseException(Request.CreateValidationErrorResponse(ModelState));
 
             // this is prone to race conditions but the service will not let us proceed anyways
-            var existing = Services.LocalizationService.GetLanguageByIsoCode(language.IsoCode);
+            var existingByCulture = Services.LocalizationService.GetLanguageByIsoCode(language.IsoCode);
 
             // the localization service might return the generic language even when queried for specific ones (e.g. "da" when queried for "da-DK")
             // - we need to handle that explicitly
-            if (existing?.IsoCode != language.IsoCode)
+            if (existingByCulture?.IsoCode != language.IsoCode)
             {
-                existing = null;
+                existingByCulture = null;
             }
 
-            if (existing != null && language.Id != existing.Id)
+            if (existingByCulture != null && language.Id != existingByCulture.Id)
             {
                 //someone is trying to create a language that already exist
                 ModelState.AddModelError("IsoCode", "The language " + language.IsoCode + " already exists");
                 throw new HttpResponseException(Request.CreateValidationErrorResponse(ModelState));
             }
 
-            if (existing == null)
+            var existingById = language.Id != default ? Services.LocalizationService.GetLanguageById(language.Id) : null;
```

Comment:
What are the reason to use `default` and not `null` ?

Reply:
Ahh I see it is `0`

Reply:
We do use a viewmodel, if it's a brand new language, then there is no ID, and therefore it is `0` because that is the default value of int.

Reply:
👍 
I know, I just think it is a code smell, to use an `int` and not `int?` when the property also have to represent the absence of a value. But that's not part of this PR, more a general issue.